### DISCUSSION TITLE
Fix bug compare two string in V3

### DISCUSSION THREAD

**Discussion by antran-pk** [spatie/laravel-activitylog#1271] [General]
**Trait: src/Traits/DetectsChanges.php**
**Func: function attributeValuesToBeLogged(string $processingEvent): array**
`$new <=> $old` will be ignore when compare two string number
Example: '0123' <=> '123', return 0;
**Solve: can check string and compare like code below**
if (gettype($old) === 'string' && $old !== $new) {
    return 1;
}

**Comment by Gummibeer** [comment]
Can't reproduce your statement/report.

https://3v4l.org/kJMhK

**Reply by antran-pk** [reply]
Sorry! My question is wrong, I updated the question already

Try this reproduce:
https://3v4l.org/nucIs
